packages/shared/src/hooks/useWebVitals.ts

```diff
@@ -0,0 +1,29 @@
+import { useCallback, useContext, useEffect } from 'react';
+import { onCLS, onFCP, onFID, onLCP, onTTFB } from 'web-vitals';
+import AnalyticsContext from '../contexts/AnalyticsContext';
+
+export function useWebVitals(): void {
+  const { trackEvent } = useContext(AnalyticsContext);
+
+  const trackMetric = useCallback(
+    (metric) => {
+      const { name, entries, id, ...rest } = metric;
+      trackEvent({ event_name: name, extra: JSON.stringify(rest) });
+    },
+    [trackEvent],
+  );
+
+  // Have to wait for document to load and ensure it's not run on SSR
+  useEffect(() => {
+    // We apply sampling rate of 10%
+    if (Math.random() < 0.1) {
```

Comment:
Maybe sampling rate through `env` so we can modify without code changes?

Reply:
I can do, but it's tricky with extension vs webapp in that case.
So not sure it actually adds much.

Reply:
We do have the option to split inside data analysis between webapp/extension
We can evaluate if it adds anything basically.

Good point though.
Let's try for now and see what data suggests.